lib/Dialect/TritonGPU/Transforms/HoistTMEMAlloc.cpp

```diff
@@ -92,6 +94,70 @@ class RemoveUnusedTMEMLoad : public OpRewritePattern<TMEMTokenLoadOp> {
   }
 };
 
+class RemoveUnusedTMEMStore : public OpRewritePattern<TMEMTokenStoreOp> {
+public:
+  using OpRewritePattern::OpRewritePattern;
+
+  LogicalResult matchAndRewrite(TMEMTokenStoreOp store,
+                                PatternRewriter &rewriter) const override {
+    auto tok = store.getToken();
+    if (!tok.hasOneUse())
+      return failure();
+    auto loop = dyn_cast<scf::ForOp>(*tok.getUsers().begin());
+    if (!loop)
+      return failure();
+    auto loopTok = loop.getBody()->getArgument(
+        tok.getUses().begin()->getOperandNumber() - 2);
+    if (!loopTok.hasOneUse())
+      return failure();
+    auto mma =
+        dyn_cast<nvidia_gpu::MMAv5OpInterface>(*loopTok.getUsers().begin());
+    if (!mma)
+      return failure();
+    auto useD = dyn_cast<BlockArgument>(mma.useAccumulator());
+    if (!useD)
+      return failure();
+    auto parent = useD.getParentBlock()->getParentOp();
+    if (parent != loop)
+      return failure();
+    auto loopInit = loop.getInitArgs()[useD.getArgNumber() - 1];
+    auto val = getBoolFromConstant(loopInit);
+    if (!val)
+      return failure();
+    if (val.value() == true)
+      return failure();
+    auto loc = store.getLoc();
+    rewriter.replaceAllUsesWith(store.getToken(), store.getDep());
+    rewriter.setInsertionPointAfter(loop);
+    Value diff = rewriter.create<arith::SubIOp>(loc, loop.getUpperBound(),
+                                                loop.getLowerBound());
+    if (diff.getType().isIndex()) {
```

Comment:
we don't use index type. I would remove this.

Reply:
I hit an issue with one of the tests having index values in the loop bounds. Would you like me to also change that test to i32?

Reply:
yes that would be great